src/app/organization/offices/offices.component.html

```diff
@@ -60,3 +66,127 @@
   </div>
 
 </div>
+
+<!-- Tree View -->
+
+<div class="container" [hidden]="viewGroup.value !== 'treeView'">
+
+  <div class="m-b-20" fxLayout="row" fxLayoutAlign="start" fxLayoutGap="20px">
+    <button mat-raised-button (click)="nestedTreeControl.expandAll()">  <!-- Bug(angular-material): https://github.com/angular/material2/issues/12170 -->
+      Expand All
+    </button>
+    <button mat-raised-button (click)="nestedTreeControl.collapseAll()">
+      Collapse All
+    </button>
```

Comment:
@Rishabh106 Everything looks great and fine. Please look over the comment in the [link](https://github.com/angular/components/issues/12170#issuecomment-406530386). As if you go through the test cases written for that particular component of nestedTreeControl, you can see `treeControl.datanodes` have been provided with a value which helps for collapseAll() and expandAll().  Give it a try if it works. Let me know regarding teh updates.

Reply:
In which file do I need to add the dataNodes value? (Sorry if its too obvious but I'm still a beginner at Angular)

Reply:
Yaa sure no problem. Feel free to ask any doubt.
You need to add another key `datanodes` for the `treeControl` in the `office.component.ts` file.

Reply:
I got it working, should I push the changes?

Reply:
Can you make another commit for that, will review and if any problem we can revert back? If everything goes fine, you can rebase it.

Reply:
Done, I'll resolve the merge conflicts after everything else is fine.

Reply:
It's working well, you can rebase it.